queue/src/main/java/org/killbill/queue/DBBackedQueue.java

```diff
@@ -521,13 +521,15 @@ public Long apply(final T input) {
         return entries;
     }
 
-
     private long getNbReadyEntries() {
         final Date now = clock.getUTCNow().toDate();
-        final String owner = CreatorName.get();
-        return sqlDao.getNbReadyEntries(now, owner, config.getTableName());
+        return getNbReadyEntries(now);
     }
 
+    public long getNbReadyEntries(final Date now) {
+        final String owner = config.getPersistentQueueMode() == PersistentQueueMode.POLLING ? null : CreatorName.get();
+        return sqlDao.getNbReadyEntries(now, owner, config.getTableName());
```

Comment:
In the case of `STICKY_EVENTS` we should be able to get info from the `inflightEvents` queue instead of doing the query?

Reply:
Yes, but only if the queue didn't overflow. It's a rare case but since we only perform the query once every 5 minutes, not sure the optimization is worth it?